src/util/decode_encode_muxed_account.js

```diff
@@ -35,27 +29,30 @@ export function decodeAddressToMuxedAccount(address, supportMuxing) {
 }
 
 /**
- * Converts an xdr.MuxedAccount to its string representation.
+ * Converts an xdr.MuxedAccount to its StrKey representation.
  *
- * By default, this returns its "G..." string representation (i.e. forcing the
- * ed25519 representation), but can return the "M..." representation via opt-in.
+ * By default, this returns its "M..." string representation (SEP-23), but it
+ * can return the "G..." representation (i.e. forcing the ed25519
+ * representation) via an explicit flag. The latter option is useful when you
+ * have an object but only care about representing its underlying public key.
  *
  * @function
  *
- * @param   {xdr.MuxedAccount} muxedAccount   account to stringify
- * @param   {bool}            [supportMuxing] converts the object into its full,
- *     proper M... address, encoding both the underlying G... address and the
- *     muxing ID, but *ONLY* when the ID is present.
+ * @param   {xdr.MuxedAccount} muxedAccount   Raw account to stringify
+ * @param   {bool} [forceEd25519=false]       If this is set to `true`, this
+ *     will always return the underlying G... address of the `muxedAccount`.
  *
  * @returns {string}  stringified G... (corresponding to the underlying pubkey)
  *     or M... address (corresponding to both the key and the muxed ID)
  */
-export function encodeMuxedAccountToAddress(muxedAccount, supportMuxing) {
+export function encodeMuxedAccountToAddress(muxedAccount, forceEd25519) {
```

Comment:
Do we need to keep the forceEd25519? This function is named `encodedMuxedAccountToAddress`, and the only valid address for a muxed account is a M address. If someone wants to encode the base account of a muxed accounmt to a G address, they should probably call mux.baseAccount() and go from there?

Is this required for compatibility?

Reply:
It's a separation of concerns: these functions are for operating between `xdr.MuxedAccount` <--> `string`, and then there's also the higher-level `MuxedAccount` object that implements the `Account` interface.

I touched on this a bit above (https://github.com/stellar/js-stellar-base/pull/485#issuecomment-1009347554), but basically we need to be able to forceEd25519 when actually building a `MuxedAccount` object from an M-address in order to get its composite G-address and ID.

Reply:
Ah I see. I still think we should follow a deconstruct and a construct approach, rather than a parameter that makes a construction behave differently. By deconstruct I mean get the base address/account id out of the xdr.MuxedAccount. We should probably have a function on MuxedAccount to do this if we don't already.

Reply:
I wrote up an overly-verbose comment on the design decisions I was making and all of the abstraction layers we need to deal with (the TL;DR is that we have `xdr.MuxedAccount` :left_right_arrow: M-address, `xdr.MuxedAccount` :left_right_arrow: G-address, and `MuxedAccount` :left_right_arrow: M-address), but I think the latest commit explains all that is needed: 10f14a377b049c1e6d94617cea4e8df16f2d325b.

I've dropped the `forceEd25519` flag and moved that subset of functionality to a new helper,
```ts
function extractBaseAddress(mAddress: string): string;
```

Does that help?